_posts/2016-05-24-journal-of-a-junior-bem.md

```diff
@@ -0,0 +1,79 @@
+
+###So what exactly is BEM?
+
+BEM is a strict way of arranging your CSS into independent and informative modules. It usually follows this pattern:
+
+<pre><code class="scss">
+.block{}
+.block__element{}
+.block--modifier{}
+
+</code></pre>
+<br>
+A block represents an object on your website, and could be a form, header, or navbar for example.
+An element represents a descendant of your block and is indicated by beginning with a double underscore ‘__ ‘. Elements are what make up your block and so can be buttons, textboxes or media items.
+A modifier is a different version of your element or block and is indicated by double hyphen ‘--’. It could be something like --active that would give a style like background-color: blue. It’ll indicate that the object would be active (yes, really!). It can also identify states like booleans, or values like bold, normal or thin for properties like font weight.
+
+The reason for double underscores and hyphens is because if your block or element has a hyphen in its name then you don’t want to confuse the name with a modifier. Like if you had a block called nav-link, it’d be BEM’d up like:
+
+<pre><code class="scss">
+.nav-link {} /* Block */
+.nav-link__field {}
+.nav-bar__nav-link--active {}
+
+</code></pre>
+<br/>
+
+###What are the benefits of using BEM compared to my previous approach?
+
+My previous approach to stylesheets was very...I’m going to go with amateur. I enjoyed writing them but I had no structure to follow. Everything was thrown into one stylesheet and there was a lot of specifics and nesting which I have now learned isn’t a fantastic idea. If I was to change an ID, then I could potentially ruin the style, if I was to change the class of a parent element I could throw off the whole section. There was tons of nesting and so a lot of repeated styles. This lead to a very confusing and daunting stylesheet (but I was proud of it anyways because “Look at all this cool stuff I wrote!”).
+
+The beautiful thing with BEM is that it eliminates a lot of those problems - even though it’s kinda ugly on the outside. It doesn’t need nesting, because a BEM CSS class is independent and doesn’t need tags or IDs to show who it is! It depends on its unique class and its modifiers. 
+
+Another nice quality about BEM is that it’s very informative, you can tell what the style is just by looking at the class. Let’s go with the analogy of a cat:
+
+<pre><code class="scss">
+.cat {}
+.cat__ear{}
+.cat--fat{}
+.cat--fat__ear{}
```

Comment:
I don't think it's wrong, it's just an alternative way of doing it. This is the page I did some of my research from
http://csswizardry.com/2013/01/mindbemding-getting-your-head-round-bem-syntax/
if it's really that wrong i'll change it though. This is my first exposure to BEM so i'm not an expert I only know what I've researched